modules/geolocation/camera_properties.py

```diff
@@ -0,0 +1,257 @@
+
+        if fov_y <= 0.0 or fov_y >= np.pi:
+            return False, None
+
+        u_scalar = np.tan(fov_x / 2)
+        if not np.isfinite(u_scalar):
+            return False, None
+
+        v_scalar = np.tan(fov_y / 2)
+        if not np.isfinite(v_scalar):
+            return False, None
+
+        return True, CameraIntrinsics(
+            cls.__create_key,
+            resolution_x,
+            resolution_y,
+            u_scalar,
+            v_scalar,
+        )
+
+    def __init__(self,
+                 class_private_create_key,
+                 resolution_x: int,
+                 resolution_y: int,
+                 u_scalar: float,
+                 v_scalar: float):
+        """
+        Private constructor, use create() method
+        """
+        assert class_private_create_key is CameraIntrinsics.__create_key, "Use create() method"
+
+        self.resolution_x = resolution_x
+        self.resolution_y = resolution_y
+
+        self.__vec_c = np.array([1.0,      0.0,      0.0], dtype=np.float32)
+        self.__vec_u = np.array([0.0, u_scalar,      0.0], dtype=np.float32)
+        self.__vec_v = np.array([0.0,      0.0, v_scalar], dtype=np.float32)
+
+    @staticmethod
+    def __pixel_vector_from_image_space(pixel: int,
+                                        resolution: int,
+                                        vec_base: np.ndarray) -> "tuple[bool, np.ndarray | None]":
+        """
+        Get u or v vector from pixel coordinate.
+        """
+        if pixel < 0:
+            return False, None
+
+        if resolution < 0:
+            return False, None
+
+        if pixel > resolution:
+            return False, None
+
+        if not is_vector_r3(vec_base):
+            return False, None
+
+        # Scaling factor with translation: 2 * p / r - 1 == (2 * p - r) / r
+        # Codomain is from -1 to 1 inclusive
+        scaling = float(2 * pixel - resolution) / float(resolution)
+
+        vec_pixel = scaling * vec_base
+
+        return True, vec_pixel
+
+    def camera_space_from_image_space(self,
+                                      pixel_x: int,
+                                      pixel_y: int) -> "tuple[bool, np.ndarray | None]":
+        """
+        Pixel in image space to vector in camera space.
+        """
+        if pixel_x < 0:
+            return False, None
+
+        if pixel_y < 0:
+            return False, None
+
+        result, vec_pixel_u = self.__pixel_vector_from_image_space(
+            pixel_x,
+            self.resolution_x,
+            self.__vec_u,
+        )
+        if not result:
+            return False, None
+
+        result, vec_pixel_v = self.__pixel_vector_from_image_space(
+            pixel_y,
+            self.resolution_y,
+            self.__vec_v,
+        )
+        if not result:
+            return False, None
+
+        # Get Pylance to stop complaining
+        assert vec_pixel_u is not None
+        assert vec_pixel_v is not None
```

Comment:
Is it possible that when we check if `vec_pixel_u` or `vec_pixel_v` is not None we return `False, None` instead of using an assertion. I bring this up because if the assertion fails then we stop the airside system in flight so it would be nice if we can handle the error instead.

Reply:
Yes, these assertions should never fail as they are guarded by line 190 above. They are to get the IDE linter to stop complaining about the variables possibly having a type of None and therefore being unusable.